src/goLanguageServer.ts

```diff
@@ -143,6 +144,44 @@ export async function registerLanguageFeatures(ctx: vscode.ExtensionContext) {
 						return null;
 					}
 					return next(document, token);
+				},
+				provideCompletionItem: (
+					document: vscode.TextDocument,
+					position: vscode.Position,
+					context: vscode.CompletionContext,
+					token: vscode.CancellationToken,
+					next: ProvideCompletionItemsSignature
+				) => {
+						const editorParamHintsEnabled = vscode.workspace.getConfiguration('editor.parameterHints', document.uri)['enabled'];
+						const goParamHintsEnabled = vscode.workspace.getConfiguration('[go]', document.uri)['editor.parameterHints.enabled'];
+
+						var paramHintsEnabled: boolean = false;
+						if (typeof goParamHintsEnabled === 'undefined') {
+							paramHintsEnabled = editorParamHintsEnabled;
+						} else {
+							paramHintsEnabled = goParamHintsEnabled;
+						}
+						var cmd: Command = undefined;
+						if (paramHintsEnabled) {
+							cmd = {title: 'triggerParameterHints', command: 'editor.action.triggerParameterHints'};
+						}
+						
+						function configureCommands(
+							r: vscode.CompletionItem[] | vscode.CompletionList | null | undefined):
+							vscode.CompletionItem[] | vscode.CompletionList | null | undefined {
+							if (r) {
+								(Array.isArray(r) ? r : r.items).forEach((i: vscode.CompletionItem) => {
+									i.command = cmd;
+								});
+							}
+							return r;
+						}
+						const isThenable = <T>(obj: vscode.ProviderResult<T>): obj is Thenable<T> => obj && (<any>obj)['then'];
```

Comment:
What does this do? I'm totally confused at this point :smile:

Reply:
vscode.ProviderResult can be a simple T or Thenable<T>. So, if the result of `next` call is a promise (thenable that has 'then' property), we chain configureCommands call using 'then'.
If the result of `next` is not thenable, it's not a promise and there is no 'then'. So, simply invoke configureCommands and pass the result of the next.